jobs/kpi-forecasting/kpi_forecasting/validator.py

```diff
@@ -0,0 +1,153 @@
+        input_table_list = []
+        config_file_list = list(self.config_data.keys())
+        for config_data in self.config_data.values():
+            # get segment data
+            metric_hub_data = config_data.metric_hub.toDict()
+            if "segments" in metric_hub_data:
+                segment_data = metric_hub_data["segments"]
+                segment_data_list.append(segment_data)
+            else:
+                segment_data_list.append(None)
+
+            # get input table info
+            input_table_list.append(config_data.write_results.toDict())
+
+        input_table_data = input_table_list.pop(0)
+        input_table_matches_first = [input_table_data == el for el in input_table_list]
+        if not all(input_table_matches_first):
+            config_file_list_string = " ".join(config_file_list)
+            raise Exception(
+                f"Input Table Data Does not all match for config list: {config_file_list_string}"
+            )
+
+        input_project = input_table_data["project"]
+        input_dataset = input_table_data["dataset"]
+        input_table = input_table_data["table"]
+
+        input_table_full = f"{input_project}.{input_dataset}.{input_table}"
+
+        segment_data = segment_data_list.pop(0)
+        segment_data_matches_first = [segment_data == el for el in segment_data_list]
+        if not all(segment_data_matches_first):
+            config_file_list_string = " ".join(config_file_list)
+            raise Exception(
+                f"Dimension Data Does not all match for config list: {config_file_list_string}"
+            )
+
+        if segment_data:
+            # this is the case where dimensions are present
+            # we only need the column names for the query
+            dimension_list = list(segment_data.keys())
+        else:
+            dimension_list = []
+
+        self.input_table_full = input_table_full
+        self.dimension_list = dimension_list
+
+    def query(self) -> str:
+        """Creates a query used to generate the validation table.
+
+        Returns:
+        (str): Query to generate validation table"""
+        if len(self.dimension_list) > 0:
+            segment_comma_separated = "," + ",".join(self.dimension_list)
+        else:
+            segment_comma_separated = ""
+        # in actual_deduped, the value for historical data won't change so we can use any_value without checking forecast_trained_at
+        compare_ctes = f"""WITH forecast_deduped as (SELECT submission_date, metric_alias {segment_comma_separated},
+                                                    MAX(forecast_trained_at) as forecast_trained_at,
+                                                    ANY_VALUE(value HAVING MAX forecast_trained_at) as forecast_value 
+                                                    FROM {self.input_table_full} 
+                                                    WHERE source='forecast' AND aggregation_period='day'
```

Comment:
Here, we might want to include validation for the `'month'` aggregation period, too. Would even be better to grab the `aggregation` value in the config `summarization` section to decide which aggregation periods should be validated over, then include `aggregation_period` as a dimension in the final validation table.

Reply:
I think it can be even simpler than that:  We know the aggregation period is written to the "aggergation_period" column, so we can just add that column to the groupby and join and we're done.  Pushed this change